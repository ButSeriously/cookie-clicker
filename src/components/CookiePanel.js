import React from 'react';

class CookiePanel extends React.Component {
  state = {
    cookiesAmount: 0,
    cookiesPerSecond: 0,
  }

  addCookie = () => {
    this.setState(prevState => ({
      cookiesAmount: prevState.cookiesAmount + 1,
    }));
  }

  render() {
    return (
      <div>
        <h1>SkyGate Ticket</h1>
        <div>
          <p>{this.state.cookiesAmount} cookies</p>
          <p>per second: {this.state.cookiesPerSecond}</p>
        </div>
        <button onClick={this.addCookie}>
          <img
            src="https://dummyimage.com/256x256/906734/fff.jpg"
            alt="cookie.jpg"
          />
        </button>
      </div>
    );
  }
}

export default CookiePanel;
