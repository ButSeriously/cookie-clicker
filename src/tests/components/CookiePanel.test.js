import React from 'react';
import { shallow } from 'enzyme';
import CookiePanel from '../../components/CookiePanel';

test('should render CookiePanel', () => {
  const wrapper = shallow(<CookiePanel />);
  expect(wrapper).toMatchSnapshot();
});

test('should add cookie', () => {
  const wrapper = shallow(<CookiePanel />);
  wrapper.find('button').prop('onClick')();
  expect(wrapper.state().cookiesAmount).toBe(1);
});
