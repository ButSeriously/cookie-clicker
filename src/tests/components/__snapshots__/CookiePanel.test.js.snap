// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CookiePanel 1`] = `
<div>
  <h1>
    SkyGate Ticket
  </h1>
  <div>
    <p>
      0
       cookies
    </p>
    <p>
      per second: 
      0
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    <img
      alt="cookie.jpg"
      src="https://dummyimage.com/256x256/906734/fff.jpg"
    />
  </button>
</div>
`;
